### PR TITLE
Make changes necessary for pip sdist

### PR DESCRIPTION
**Pull request by lanctot**
- setup.py: Add checks for C++ compiler and python-dev headers
- setup.py: Look for requirements.txt in current and parent directory
- setup.py & noxfile.py: Add flag to determine building and testing everything as opposed to just pyspiel
- Add requirements.txt to MANIFEST.in (read by setup.py)
- Change name to `open_spiel` and version to `0.2.0`
- Use compiler specified by CXX environment variable and drop down to default of clang++

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -1,17 +1,3 @@
-# Copyright 2019 DeepMind Technologies Ltd. All rights reserved.
```
Comment (jblespiau) [inline noxfile.py:1]: Should we keep that?

File: setup.py [hunk 2]
```diff
@@ -1,17 +1,3 @@
-# Copyright 2019 DeepMind Technologies Ltd. All rights reserved.
```
Comment (jblespiau) [inline setup.py:1]: same

File: setup.py [hunk 3]
```diff
@@ -45,36 +31,70 @@ class BuildExt(build_ext):
...
 
   def run(self):
+    self._check_build_environment()
+    for ext in self.extensions:
+      self.build_extension(ext)
+
+  def _check_build_environment(self):
+    """Check for required build tools: CMake, C++ compiler, and python dev."""
     try:
       subprocess.check_call(["cmake", "--version"])
```
Comment (jblespiau) [inline setup.py:41]: Do we want a minimal version? I remember some people had issues with old Cmakes, so maybe we can check the version is higher that some version you know works (I have not been following which version works or does not, maybe you know more.

Same for Clang

**Review by jblespiau — APPROVED** [review]

File: noxfile.py (hunk 1 above)
Comment (lanctot) [reply noxfile.py:1]: Whoops, yes!

File: setup.py (hunk 2 above)
Comment (lanctot) [reply setup.py:1]: Yes!

File: setup.py (hunk 3 above)
Comment (lanctot) [reply setup.py:55]: I thought of that originally as well, but I want to shy away from it, for a few reasons:

- Checking the versions is compiler-dependent, so to do this we'd have to force clang specifically.
- Everything should still work with any C++17 compiler, so we shouldn't strictly force clang++ (right now you can override it if the CXX environment variable is set), but set clang++ to the default
- I actually don't know what version we really _need_ but I know what works: Travis Ubuntu 18.04 has Clang 7.0.0 installed. So I modified the comment to say specifically what we need (a C++17 compliant compiler, but that we recommend Clang >= 7.0.0).

What do you think?

**Pull request merged**